### PR TITLE
fix: Retrieve chained message if it doesn't exist locally yet.

### PR DESCRIPTION
**Pull request by lan17**
https://leanplum.atlassian.net/browse/LP-6124?filter=-1

Tested in staging by setting up a case where:

  * App is already running.
  * On dashboard I create a new In-App message that is active.
  * Create a push message with open existing message pointing to the just created in-app message.
  * Send out Push as preview, and have the test app display in-app message after i open the push notification.

### CODE REVIEW

**Comment by Enuviel** [comment]
test this please

File: AndroidSDK/src/com/leanplum/LeanplumPushService.java [hunk 1]
```diff
@@ -383,7 +383,33 @@ private static void showNotification(Context context, Bundle message) {
     }
 
     try {
-      notificationManager.notify(notificationId, builder.build());
+      // Check if we have a chained message, and if it exists in var cache.
```
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/LeanplumPushService.java]: could you please create a method for this?

Comment (lan17) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: Where should it go?  Same file?

Comment (Enuviel) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: ActionContext.java ?

Comment (lan17) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: The logic here works on `Bundle` and ActionContext.java deals with `Map`.  Would that still be appropriate?

Comment (lan17) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: I extracted teh logic that works on just the action map to ActionContext.

File: AndroidSDK/src/com/leanplum/ActionContext.java [hunk 2]
```diff
@@ -406,27 +406,55 @@ public void variablesChanged() {
...
+        });
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Extract chained messageId from parent message's actionMap.  If it doesn't exist then return null.
+   */
+  public static String getChainedMessageId(Map<String, Object> actionMap) {
```
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/ActionContext.java]: please remove public

File: AndroidSDK/src/com/leanplum/LeanplumPushService.java (hunk 1 above)
Comment (Enuviel) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: you can pass String  to this method - String actionData = message.getString(Constants.Keys.PUSH_MESSAGE_ACTION) and return true/false

**Comment by lan17** [comment]
Added unit tests and moved things around :)

File: AndroidSDK/src/com/leanplum/LeanplumPushService.java (hunk 1 above)
Comment (lan17) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: fixed

Comment (lan17) [reply AndroidSDK/src/com/leanplum/LeanplumPushService.java]: fixed

**Pull request merged**